sklearn/utils/_array_api.py

```diff
@@ -460,6 +480,39 @@ def _weighted_sum(sample_score, sample_weight, normalize=False, xp=None):
         return float(xp.sum(sample_score))
 
 
+def _isneginf(X):
+    xp, _ = get_namespace(X)
+    is_inf = xp.isinf(X)
+    signbit = xp.sign(X) < 0
+    return xp.logical_and(is_inf, signbit)
+
+
+def _nanmin(X, axis=None):
+    # TODO: refactor once nan-aware reductions are standardized:
+    # https://github.com/data-apis/array-api/issues/621
+    xp, _ = get_namespace(X)
+    if _is_numpy_namespace(xp):
+        return xp.asarray(numpy.nanmin(X, axis=axis))
+
+    else:
+        X = xp.min(xp.where(~xp.isnan(X), X, xp.asarray(+xp.inf)), axis=axis)
+        # Replace Infs from all NaN slices with NaN again
+        return xp.where(~xp.isinf(X), X, xp.asarray(xp.nan))
```

Comment:
This would replace a valid `inf` if the input was actually all inf:

```python
import numpy as np

X = np.array([np.inf, np.inf])
np.nanmin(X)
# inf
```

Reply:
Fixed. The code now is much better I think, and it has tests!